view/frontend/web/js/view/minicart/fast-button.js

```diff
@@ -23,14 +23,48 @@ define([
         observableProperties: [
             'items'
         ],
-        initialize: function () {
-            var self = this,
+        initialize: function () { 
+            var self = this, 
                 minicart = $('[data-block="minicart"]');
             this._super();
+            self.cartId = ko.observable('');
+            self.fastAppId = ko.observable(fastConfig.getAppId());
             self.shouldShowFastButton = ko.observable(fastConfig.shouldShowFastOnCart());
-            self.fastDark = ko.observable(fastConfig.getBtnTheme());
+            self.fastDark = ko.observable(fastConfig.getBtnTheme() === 'dark');
+
+            function ajaxCall(callback){
+                $.ajax({
+                    url: '/fast/config/fast',
+                    type: 'GET',
+                    dataType: 'json'
+                }).done(function(data){
+                    self.cartId(data.cartId);
+                    self.fastAppId(data.appId);
+                    self.fastDark(data.theme === 'dark');
+                    callback(data);
+                }).fail(function(data){
+                    callback(null);
+                });
+            };
+            if((!self.fastAppId() || !self.cartId()) 
+                && customerData.get('cart')().items && customerData.get('cart')().items.length > 0){
+                //initial cart id lookup on page load
+                ajaxCall(function(data){
+                    if(data == null){ 
+                        console.error('Config call failed');
```

Comment:
Is this needed ?

Reply:
trying to cover all scenarios to set fast configs , in case page refresh and no cart updates where done. i'm still looking in to combine 2 calls in to 1 , this will remove this aja call all together

Reply:
removed error log, updated to have callback as optional